### PR TITLE
Add version guards around Transform hidden index settings

### PR DESCRIPTION
**Pull request by AthenaEryma**
This commit ensures that the hidden index settings are only applied to the
Transform index templates when the cluster can support those settings.

Also unmutes the tests which were failing.

Fixes https://github.com/elastic/elasticsearch/issues/53931

---

I've run the tests several times locally and had no failures that I can attribute to the code in question here (though I have had a few that I believe are the fault of my environment, so I'll likely re-run BWC in CI a few times to be sure).

### CODE REVIEW

**Comment by gwbrown** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by gwbrown** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by gwbrown** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by gwbrown** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformInternalIndex.java [hunk 1]
```diff
@@ -96,19 +99,22 @@ public static IndexTemplateMetaData getIndexTemplateMetaData() throws IOExceptio
         return transformTemplate;
     }
 
-    public static IndexTemplateMetaData getAuditIndexTemplateMetaData() throws IOException {
+    public static IndexTemplateMetaData getAuditIndexTemplateMetaData(Version compatabilityVersion) throws IOException {
```
Comment (hendrikmuhs) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformInternalIndex.java]: spelling nit: `compatabilityVersion` -> `compatibilityVersion`

**Review by hendrikmuhs — APPROVED** [review]
LGTM

@elastic/ml-qa May I ask you to check that job messages are working (not missing) in the UI when you do release testing for 7.7, both for a fresh install as well as for an upgraded cluster (7.6 -> 7.7)?

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformInternalIndex.java [hunk 2]
```diff
@@ -399,7 +405,7 @@ protected static void installLatestAuditIndexTemplateIfRequired(
 
         // Installing the template involves communication with the master node, so it's more expensive but much rarer
         try {
-            IndexTemplateMetaData indexTemplateMetaData = getAuditIndexTemplateMetaData();
+            IndexTemplateMetaData indexTemplateMetaData = getAuditIndexTemplateMetaData(clusterService.state().nodes().getMinNodeVersion());
```
Comment (hendrikmuhs) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformInternalIndex.java]: FWIW, for later readers: I think this is the important change of this PR as this is the place where a (potentially new) non-master calls the (potentially old) master.

**Review by droberts195 — APPROVED** [review]
LGTM

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformInternalIndex.java (hunk 1 above)
Comment (gwbrown) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformInternalIndex.java]: Thanks, good catch.

**Comment by gwbrown** [comment]
Thanks for the reviews folks!

**Pull request merged**